app/experimenter/legacy-ui/rapid/components/experiments/ResultsTable.tsx

```diff
@@ -0,0 +1,270 @@
+  // This is used in conjunction with resultsData as an ordered
+  // list of metrics to display in the table from left to right
+  const resultsMetricsList = [
+    {
+      value: "retained",
+      name: "2-Week Browser Retention",
+      tooltip: RESULT_COLUMN_TIPS.RETENTION,
+    },
+    {
+      value: "search_count",
+      name: "Daily Mean Searches Per User",
+      tooltip: RESULT_COLUMN_TIPS.SEARCH,
+    },
+    {
+      value: "identity",
+      name: "Total Users",
+      tooltip: RESULT_COLUMN_TIPS.USER_COUNT,
+    },
+  ];
+
+  const featureNameMappings = getFeatureNameMappings();
+  featureData.forEach((feature) => {
+    const featureMetricID = `${feature}_ever_used`;
+    resultsMetricsMap[featureMetricID] = "binomial";
+    resultsMetricsList.unshift({
+      value: featureMetricID,
+      name: `${featureNameMappings[feature]} Conversion`,
+      tooltip: RESULT_COLUMN_TIPS.CONVERSION,
+    });
+  });
+
+  return { resultsMetricsList, resultsMetricsMap };
+};
+
+const computeSignificance = (lower, upper) => {
+  if (lower < 0 && upper < 0) {
+    return Significance.NEGATIVE;
+  } else if (lower > 0 && upper > 0) {
+    return Significance.POSITIVE;
+  } else {
+    return Significance.NEUTRAL;
+  }
+};
+
+const getResultsData = (data, resultMetrics) => {
+  const analysisData = data.analysis;
+  const fullResultsData = analysisData["overall"];
+
+  const results = {};
+  let totalPopulation = 0;
+  Object.values(fullResultsData).forEach((row: AnalysisPoint) => {
+    const { metric, branch, statistic, point, lower, upper, comparison } = row;
+    results[branch] = results[branch] || {};
+
+    // `comparison` could be empty, `difference`, or `relative_uplift`.
+    // Here we want to display the empty comparison which means a given
+    // branch's values is not relative to another branch. We also want
+    // to use the `difference` comparison to decide whether there was
+    // any positive or negative significance.
+    if (metric in resultMetrics && resultMetrics[metric] === statistic) {
+      results[branch][metric] = results[branch][metric] || {};
+
+      let newData = {};
+      if (!comparison) {
+        newData = { lower, upper, point };
+      }
+
+      if (comparison === "difference") {
+        newData = { significance: computeSignificance(lower, upper) };
+      }
+
+      Object.assign(results[branch][metric], newData);
+    }
+
+    // Compute the population sum to be used for % per variant.
+    if (metric === "identity") {
+      totalPopulation += point;
+    }
+  });
+  setPopulationPercentages(results, totalPopulation);
+
+  return results;
+};
+
+const populationField = (point, percent) => {
+  return (
+    <div>
+      <div className="font-weight-bold">{point}</div>
+      <div className="subscript">{percent}%</div>
+    </div>
+  );
+};
+
+const showSignificanceField = (significance, interval) => {
+  {
+    let significanceIcon, className;
+    switch (significance) {
```

Comment:
You know there's a fun (but kinda janky) way to do exhaustive switch statements in ts with enums:

https://stackoverflow.com/a/39419171

I've used it to make sure my switch cases are exhaustive.  Might be overkill here but, if you wanna squeeze some real strong type enforcement out, might be interesting?